### ISSUE TITLE
timeout: timedout and WaitTimeoutError messages when browsing Netflix

### ISSUE THREAD

**Issue by nonobio** [CastagnaIT/plugin.video.netflix#1285]
#### Your Environment
- Netflix add-on version: 1.18.2+matrix.1
- Operating system version/name: Chrome OS 96.0.4664.111 (Build officiel) (32 bits)
- Device model:  Acer Chromebook R 13 CB5-312T-K62F

Used Operating system:
* [x] Android (Android app support but the operating system is "Chrome OS")
* [ ] iOS
* [ ] Linux
* [ ] OSX
* [ ] Raspberry-Pi
* [ ] Windows

### Describe the bug

Hello,

After I installed the add-on, I chose to connect with the authentication key: it said "timeout: timedout"..., but then "Connexion success". So I tried to launch the add-on but i have again the "timeout: timedout" message.
I searched for a solution on Github and found it could be related to the "IPC" setting. It was enabled by default so I disabled it.

Now it is better: when I launch the add-on I can see my profiles, but the first time I choose one and try to enter into it: "WaitTimeoutError". I press OK to close the error. I try again: it works. But every new level I try to enter into: same error the first time.

I tried to log-off, restore all settings to defaults, uninstall and reinstall the add-on: I have the same issue.

Thanks.

#### Expected behavior

Browse Netflix without timeout errors.

#### Actual behavior

I have a lot of timeout errors.

#### Steps to reproduce the behavior

1. Open the add-on
2. Browse Netflix

### Debug log

The debug log can be found here:

https://paste.kodi.tv/ujulojequf.kodi

[### Additional context or screenshots (if appropriate)

#### Other information

Actually I have two issues with the Netflix add-on I just installed on my Chromebook, this is the second:
https://github.com/CastagnaIT/plugin.video.netflix/issues/1286

**Comment by CastagnaIT** [comment]
for WaitTimeoutError error: https://github.com/CastagnaIT/plugin.video.netflix/wiki/FAQ-%28Errors%29#addon-signals-call-timeout

**Comment by nonobio** [comment]
Hi @CastagnaIT :) ,

Thanks for your reply.

> Addon Signals call timeout
> This can happen most of the times when:
> 
> Could be a problem of conflicts with other add-ons who are performing other operations at the same time with AddonSignals add-on.

I only installed the Netflix Add-on. Actually, I just installed Kodi on my Chromebook only to use Netflix on it ;) .
So it is a fresh installation but I used my usual advancedsettings file to connect to my MYSQL database, in case I need to use Kodi for other stuff (play sources on my local network).

Do you think it could be related to the connexion to my MYSQL database ? If yes, I could delete my advancedsettings file for this device...

> The only solution is try to enable IPC over HTTP: In add-on expert settings, turn on Enable IPC over HTTP.

I did that.

> Your connection/ISP is not working optimally or Netflix server fails
> Not solvable here.

The add-on works fine on my other devices on the same connection/ISP (Windows PC and LibreElec PC).
It could be related to the Wifi connexion of my Chromebook but I brought her closer to my router so I don't think it is the problem...

> Your device uses a memory card or disk that is too slow.

It is the internal memory of my Acer Chromebook R 13 CB5-312T-K62F (64 Go eMMC) : could it be too slow ?

> In rare cases it may represent an internal error of the addon.

Ok, but do you see anything in my log who help to locate the issue ?

**Comment by CastagnaIT** [comment]
> I did that.

from the log:
```
File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.module.addon.signals/lib/AddonSignals.py", line 115, in waitForReturn
raise WaitTimeoutError
                                                   AddonSignals.WaitTimeoutError
```
means that IPC over HTTP is not enabled

**Comment by nonobio** [comment]
Hi @CastagnaIT :),

Sorry for my late reply.

> means that IPC over HTTP is not enabled

Yes but when it was enabled (by default), it was worst as I said:

> After I installed the add-on, I chose to connect with the authentication key: it said **"timeout: timedout"**..., but then "Connexion success". So I tried to launch the add-on but i have again the "timeout: timedout" message.
> I searched for a solution on Github and found it could be related to **the "IPC" setting. It was enabled by default so I disabled it.**
> 
> **Now it is better: when I launch the add-on I can see my profiles,** but the first time I choose one and try to enter into it: "WaitTimeoutError". I press OK to close the error. I try again: it works. But every new level I try to enter into: same error the first time.

I will try to enable it again.

Thanks

**Comment by tniagcpm** [comment]
Hi @nonobio,
I have seen this on 2 systems - a new Kodi install on Linux, and a previous OSMC install on a Pi3.
Same as you - a 30sec delay, a timeout error, and the next try works without delay.

Do you have IPv6 connectivity?
I do, and once I disable IPv6 on the systems the delays stop.

I am still investigating why it makes a difference.
From some wireshark captures, perhaps a mix of both IPv4 & IPv6 connections to Netflix cause it?

**Comment by tniagcpm** [comment]
Hi @nonobio ,

My problems were due to some ICMPv6 packets being blocked.
Kodi+plugin makes alternate IPv4 & IPv6 connections. IPv4 works, with no delay, IPv6 fails after a 30sec delay.

My router has a NFTables firewall with these forwarding rules:
> meta l4proto ipv6-icmp accept
> ct state invalid drop
> ct state related,established accept

Problem was the "ct state invalid drop" was **before** "... ipv6-icmp accept".
My guess is that it was interfering with path MTU discovery.

Anyway, with the change above, there are no delay problems on either of my Kodi systems.

BTW, I tried switching IPv4 off on the Kodi system, but Kodi fails since it tries to detect an Internet connection using:
> www.w3.org
> www.msftncsi.com

but those sites do not have an IPv6 address, so Kodi errors with No Internet Connection...

**Comment by CastagnaIT** [comment]
it seems that the localhost packages sent by addons via IPC are blocked,
it is possible that the chromebook disallow the access to the localhost IP to route packets around kodi
you should do a search for possible problems on chromebook and localhost usage
or/else for some localhost settings to change or something similar,
another possibility (but should not) is that chromebook use a different IP for localhost, usually by default should be 127.0.0.1

**Comment by nonobio** [comment]
Hi @tniagcpm and @CastagnaIT ,

I start checking about IPv6: there is an IPv6 address in my Chromebook network settings, but I don't see how to disable it.

Then, I just tried to connect on my second Wifi network: 2,4 GHz (instead of the 5 GHz from my same router.): no delay on Netflix Kodi !

So I switched back to the 5 GHz Wifi: no delay!

Maybe it is just temporary or maybe there was an issue with the Wifi connexion, I don't know but for now, it works.

So I didn't test your others suggestions (NFTables firewall and localhost). I will if the issue comes back.

Thanks :)
